### ISSUE TITLE
How to use initialize_for_dotnet_command_line_with_args with command line args?

### ISSUE THREAD

**Issue by AArnott** [OpenByteDev/netcorehost#31]
I can't figure out how to call `initialize_for_dotnet_command_line_with_args` with `env::args()`. I need to translate the `String` that `args()` produces into a `PdCstr`, but the latter seems to want fixed strings known at compile-time. I'm very new to rust, so I apologize if this comes off as a silly question. Your help is appreciated!

Here is what I've tried:

```rs
use std::env;

use netcorehost::{nethost, pdcstr, pdcstring::PdCStr};

fn main() {
    let hostfxr = nethost::load_hostfxr().unwrap();

    // Pass all arguments to the .NET Core application
    let context = hostfxr.initialize_for_dotnet_command_line_with_args(env::args().skip(1).into_iter().map(|f| pdcstr!(f)).collect()[..]).unwrap();
    let result = context.run_app().as_hosting_exit_code().unwrap();
    println!("Exit code: {}", result);
}
```

```
    Checking dotnet-sdk-matching-runtime v0.1.0 (...)
error[E0435]: attempt to use a non-constant value in a constant
  --> src\main.rs:10:120
   |
10 |     let context = hostfxr.initialize_for_dotnet_command_line_with_args(env::args().skip(1).into_iter().map(|f| pdcstr!(f)).collect()[..]).unwrap();
   |                                                                                                                --------^-
   |                                                                                                                |       |
   |                                                                                                                |       non-constant value
   |                                                                                                                help: consider using `let` instead of `const`: `let _WIDESTRING_U16_MACRO_UTF8`
```

**Comment by OpenByteDev** [comment]
You will want to use [PdCString::from_os_str](https://docs.rs/netcorehost/latest/netcorehost/pdcstring/struct.PdCString.html#method.from_os_str) in this case instead of the macro.

**Comment by OpenByteDev** [comment]
Published in 0.15.1
